third-party/realdds/src/dds-stream-server.cpp

```diff
@@ -18,9 +19,15 @@ using namespace eprosima::fastdds::dds;
 using namespace realdds;
 
 
-dds_stream_server::dds_stream_server( std::shared_ptr< dds_topic_writer > const & writer )
-    : _writer( writer )
+dds_stream_server::dds_stream_server( std::string const & name,
+                                      dds_stream_profiles const & profiles,
+                                      int default_profile_index )
+    : _name( name )
+    , _profiles( profiles )
+    , _default_profile_index( default_profile_index )
```

Comment:
Should also check if `default_profile_index` is in [0, profiles.size()-1] range

Reply:
Added.